src/cartservice/Dockerfile

```diff
@@ -1,28 +1,35 @@
-FROM microsoft/dotnet:2.1-sdk-alpine as builder
+FROM gcr.io/dotnet-debugger/aspnetcore:2.1 as builder
 WORKDIR /app
 COPY . .
 RUN dotnet restore && \
     dotnet build && \
-    dotnet publish -c release -r linux-musl-x64 -o /cartservice
+    dotnet publish -c debug -o /cartservice
 
 # cartservice
-FROM alpine:3.8
+FROM gcr.io/dotnet-debugger/aspnetcore:2.1
+
+# Dependencies for runtime
+# busybox-extras => telnet
+RUN apt-get install -q -y --no-install-recommends \
```

Comment:
we had "busybox-extras" to have telnet (as you can see on Line 12) I see you removed a few things and changed a few things here. Also y ou can move this below the GRPC_HEALTH_PROBE downloading (as it was before).

We get some level of caching from downloading grpc health probe first.